mps_youtube/players/mplayer.py

```diff
@@ -0,0 +1,227 @@
+import os
+import sys
+import tempfile
+import subprocess
+import re
+
+from .. import g, screen, c, paths, config, util
+
+from ..player import CmdPlayer
+
+mswin = os.name == "nt"
+not_utf8_environment = mswin or "UTF-8" not in sys.stdout.encoding
+
+
+class mplayer(CmdPlayer):
+    def _generate_real_playerargs(self):
+        """ Generate args for player command.
+
+        Return args.
+
+        """
+
+        if "uiressl=yes" in self.stream['url'] and "mplayer" in config.PLAYER.get:
+            ver = g.mplayer_version
+            # Mplayer too old to support https
+            if not (ver > (1, 1) if isinstance(ver, tuple) else ver >= 37294):
+                raise IOError("%s : Sorry mplayer doesn't support this stream. "
+                              "Use mpv or update mplayer to a newer version" % self.song.title)
+
+        args = config.PLAYERARGS.get.strip().split()
+
+        known_player = util.is_known_player(config.PLAYER.get)
```

Comment:
We already know what the player is once this class is being run, right?

`util.is_known_player()` probably needs to be changed or removed with this refactor.

Reply:
Done!